### ISSUE TITLE
Host(s) already exist in Satellite6

### ISSUE THREAD

**Issue by michaelsstuff** [ahuffman/sat6-create-hosts#2]
Hi,

updated to your new update. I also upgraded satellite to 6.4 and recreated the compute ressource.
Now I get no error, but I get 
`"Host(s) already exist in Satellite6:",`
But the host does not exist in satellite nor in vmware...

Here is my playbook:
```
---
- name: "Satellite 6 create new hosts"
  hosts: "localhost"
  tasks:
    - name: "Provision Satellite 6 Hosts"
      include_role:
        name: "sat6_create_hosts"
      vars:
        sat6_fail_on_existing: False
        sat6_hosts:
          - name: "ctx-el7-99"
            domain: "ctlm.de"
            host_group: "Ansible/EL7"
            location: "Nürnberg"
            ipv4: "10.30.6.99"
            compute_resource: "vcsa"
            parameters:
              - name: "kt_activation_keys"
                value: "rhel7"
```
ansible-playbook run with -vv
[log.txt](https://github.com/ahuffman/sat6-create-hosts/files/3095031/log.txt)

**Comment by ahuffman** [comment]
Can you turn debugging on (-v) so I can see the output on the host existence check to determine what's going on for you?  Also can you goto Hosts>All Hosts and search for the shortname of the host you're trying to create and see if anything comes back from your search?

**Comment by ahuffman** [comment]
Nevermind on the debugging, I see you attached a log.txt, which is extremely useful in determining what's wrong.  

I can see the Role is properly skipping the domain defaulted from Host Group, which would be `mydom.de`.  This is good and expected.

I then see the set host fqdn task (domain specified) setting the fqdn to `ctx-el7-99.mydom.de` which is not expected since you specified `ctlm.de` in the role's vars.  

Is there any chance you have `sat6_hosts` defined elsewhere in a group_vars/host_vars file?

Can you add an additional debug to your playbook that calls the sat6_create_hosts role like this, prior to executing the Role?:
```
- debug:
    var: "sat6_hosts"

- include_role:
 ...
```
How the `mydom.de` value is getting to that task is a mystery unless you have `sat6_hosts` specified elsewhere in your inventory, host_vars (for localhost), group_vars, or playbook extra_vars, or even in your Role defaults somewhere.  The value you are debugging with should not be getting set looking at the logic, and the values coming from the API requests.
```
TASK [sat6_create_hosts : Set Satellite6 host fqdn | Domain Default from Host Group] ********************************************************************************************************************************************************
task path: /home/COMP/mschietzsch/ansible/roles/sat6_create_hosts/tasks/create_host.yml:43
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [sat6_create_hosts : Set Satellite6 host fqdn | Domain Specified] **********************************************************************************************************************************************************************
task path: /home/COMP/mschietzsch/ansible/roles/sat6_create_hosts/tasks/create_host.yml:50
ok: [localhost] => {"ansible_facts": {"host_fqdn": "ctx-el7-99.mydom.de"}, "changed": false}

TASK [sat6_create_hosts : debug] ************************************************************************************************************************************************************************************************************
task path: /home/COMP/mschietzsch/ansible/roles/sat6_create_hosts/tasks/create_host.yml:55
ok: [localhost] => {
    "host_fqdn": "ctx-el7-99.mydom.de"
}
```

**Comment by michaelsstuff** [comment]
Uhm.. sorry for the confusion, I tried to change the names for privacy but seem to have failed to to do it in the role :-/
ctlm.de == mydom.de

**Comment by michaelsstuff** [comment]
Also the search for ctx-el7-99 did not give back any result (any org, any location)

**Comment by ahuffman** [comment]
@MSchietzsch go ahead and test again with v1.0.3 (just released).  I'm not hitting this issue in my lab on v6.4.2.1 of Satellite6.

You should never get an issue with the host existing unless there is already a host with the exact name of `{{ host.name }}.{{ host.domain }}`:
```yaml
- name: "Set Satellite6 Host {{ host_fqdn }} Exists | True"
      set_fact:
        host_exists: True
      when:
        - "host_qry.json.results | length > 0"
        - "host_qry.json.results[0].name == host_fqdn"
```
See the code block here for that logic:  https://github.com/ahuffman/sat6-create-hosts/blob/master/tasks/create_host.yml#L43-L108

Please let me know how re-testing goes so I can close this issue if you're good.

**Comment by ahuffman** [comment]
@MSchietzsch Also, I think I finally see your problem.  In your log, I see you have no compute_profile defined in the host group (null).  You also didn't specify one in your sat6_hosts list.  So without a compute_profile_id the host creation was skipped.

Either assign a default compute profile in your host group or specify one.

The summary message was a false negative, because it should be showing a list of existing hosts in the run.  It is instead empty:
```
TASK [sat6_create_hosts : debug] ************************************************************************************************************************************************************************************************************
task path: /home/COMP/mschietzsch/ansible/roles/sat6_create_hosts/tasks/main.yml:87
ok: [localhost] => {
    "msg": [
        "Host(s) already exist in Satellite6:", 
        []
    ]
}
```
I'm not positive how that was displayed, but I don't think it'll happen anymore with some of the logic changes I've made today.

Let me know how everything works once you get the compute profile specified or defaulted in your host group.  Thanks!

**Comment by michaelsstuff** [comment]
Hi,
thans for the feedback. I updated to the current master and added the compute_profile to the playbook.
I also had to add subnet, as this is not part of the group.

```
---
- name: "Satellite 6 create new hosts"
  hosts: "localhost"
  tasks:
    - name: "Provision Satellite 6 Hosts"
      include_role:
        name: "sat6_create_hosts"
      vars:
        sat6_fail_on_existing: False
        sat6_hosts:
          - name: "ctx-el7-99"
            domain: "ctlm.de"
            host_group: "Ansible/EL7"
            location: "Nürnberg"
            ipv4: "10.99.64.99"
            subnet: "Corp Net"
            compute_resource: "vcsa"
            compute_profile: "DHS"
            parameters:
              - name: "kt_activation_keys"
                value: "rhel7"
```

This results in:
```
TASK [sat6_create_hosts : Set Satellite6 Subnet ID] *****************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"msg": "The conditional check 'subnet_qry.json.results[0].name == host.subnet' failed. The error was: error while evaluating conditional (subnet_qry.json.results[0].name == host.subnet): list object has no element 0\n\nThe error appears to have been in '/home/CTLM/mschietzsch/ansible/roles/sat6_create_hosts/tasks/create_host.yml': line 257, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: \"Set Satellite6 Subnet ID\"\n      ^ here\n"}

```
[log.txt](https://github.com/ahuffman/sat6-create-hosts/files/3120454/log.txt)

**Comment by ahuffman** [comment]
I just did a similar test to validate. I found my subnets are all named as such "192.168.122.0" and did not hit this bug.  I setup a new subnet and called it "Corp Net".  I was able to reproduce the problem with the following parameters:
```yaml
sat6_fail_on_existing: false
sat6_hosts:
  - name: testserver
    ipv4: 192.168.122.55
    comment: testing sat6 provisioning
    host_group: RHEL7/test/RHEL7-nested
    location: Garage
    domain: huffnet.org
    subnet: Corp Net
    parameters:
      - name: example_param
        value: test123
      - name: test2
        value: more data
    compute_resource: huff-fedajh.huffnet.org
    compute_profile: 3-Large
```

I found that my API queries needed quoting around the jinja2 variables:
**Before**: `api_body: "{\"search\": \"name={{ host.subnet }}\"}"`
**After**: `api_body: "{\"search\": 'name=\"{{ host.subnet }}\"'}"`

This was not a problem when using strings without spaces, but once the spaces were introduced to the query it wasn't evaluating the search properly.  I went ahead and updated all the API queries in the code with the proper quote escaping to be safe.  I ran a few quick tests in my lab and everything appears to be functioning.  

Please go ahead and see if this allows you to get your hosts created now.

**Comment by ahuffman** [comment]
Also @MSchietzsch I created a new issue for the subnet query problem #3, and closed it with the last commit.
